src/Containers/DropArea/DropArea.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useEffect } from 'react';
+import styled from 'styled-components';
+import { CloudUploadAlt } from '@styled-icons/fa-solid/CloudUploadAlt';
+import { MainInterface, Main } from '@Utils/BaseStyles';
+import { flex } from '@Utils/Mixins';
+import Dropzone, {
+    useDropzone,
+    DropEvent,
+    FileRejection,
+} from 'react-dropzone';
+import Lottie from 'react-lottie';
+import { animationData } from './animationData';
+
+const ICON_OPACITY = 0.7;
+const OPACITY_WHEN_DISABLED = 0.4;
+const ICON_HEIGHT = 60;
+
+/**
+ * options for the lottie animation that occurs instead of
+ * the icon when dragging over the dropArea component
+ */
+const lottieOptions = {
+    loop: true,
+    autoplay: true,
+    animationData,
+    rendererSettings: {
+        preserveAspectRatio: 'xMidYMid slice',
+    },
+};
+
+export interface IDropAreaProps
+    extends MainInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    message?: string;
+    buttonText?: string;
+    onClickHandler?: (this: GlobalEventHandlers, ev: MouseEvent) => any; // React.MouseEventHandler<HTMLElement> | undefined;
+    onDragEnter?: React.DragEventHandler<HTMLElement> | undefined;
+    onDragLeave?: React.DragEventHandler<HTMLElement> | undefined;
+    onDropHandler?:
+        | (<T extends File>(
+              acceptedFiles: T[],
+              fileRejections: FileRejection[],
+              event: DropEvent,
+          ) => void)
+        | undefined;
+    isDisabled?: boolean;
+    width?: number;
+}
+
+export const DropArea: React.FC<IDropAreaProps> = ({
+    message = 'Drag & Drop your files here',
+    buttonText = 'Browse Files',
+    onClickHandler = () => null,
+    onDragEnter = () => null,
+    onDragLeave = () => null,
+    onDropHandler = () => null,
+    isDisabled = false,
+    ...props
+}): React.ReactElement => {
+    const { getInputProps, getRootProps, isDragActive, inputRef } = useDropzone(
+        {
+            onDragEnter,
+            onDragLeave,
+            onDrop: onDropHandler,
+            disabled: isDisabled,
+        },
+    );
+
+    const getLottieAnimationOrIcon = (isDragEnter: boolean): JSX.Element => {
+        if (isDragEnter)
+            return (
+                <Lottie
+                    options={lottieOptions}
+                    width={140}
+                    height={ICON_HEIGHT}
+                />
+            );
+        return <Icon as={CloudUploadAlt} />;
+    };
+
+    const fireEventOnInput = () => {
+        if (inputRef.current) {
+            inputRef.current.dispatchEvent(
+                new MouseEvent('click', { bubbles: false }),
+            );
+        }
+    };
+
+    useEffect(() => {
+        if (inputRef.current) {
+            inputRef.current.onclick = onClickHandler;
+        }
+    }, [onClickHandler]);
```

Comment:
What's this for?

Reply:
this is the only way to assign onclick handler to input element

Reply:
html input already has native onClick... remove this

Reply:
it doesn't work. that's the reason I have done it like this. I already had it like you say and had to change because didn't work